### DISCUSSION TITLE
Do we need a 2.78.1 hotfix?

### DISCUSSION THREAD

**Discussion by rmarquis** [etlegacy/etlegacy#1772] [General]
Just wondering about the follow up of the recent release. Since there is at least one annoying bug that might qualify as "game breaking", releasing a hotfix soon might be a good idea - if we figure out a proper solution.

In the meantime, we probably should push only other fixes to very annoying bugs, and very small no brainer changes as to polish the release - and delay the rest (f.e. ryzyk server demo changes might be too big to ensure there is no side effects, etc).

Not sure about a specific timeframe, but looking at 2.77 and 2.77.1, just 2 or 3 extra weeks? That might be enough to collect feedback on other regressions. The usual work can continue in PR, to be merged later on.

Thoughts and suggestions?

**Comment by Aciz** [comment]
Yes, I do think the window focus issue alone is enough to warrant a hotfix release. But yeah, let's wait for a week or two still, to gather potential other issues.

**Reply by Helldembez** [reply]
For now people can use a 2.77-snapshot version for the client though, while servers still run latest. But I guess most users are not comfortable with doing that.

**Comment by rmarquis** [comment]
Quick note: now that #1762  is done, there is very little left I'd like to do before releasing the hotfix: just some minor UI cleanup.
As far as I am concerned, #1777 should ideally be taken care of too but it really depends on what feedback we can collect, as nobody seems to have a real clue here. It's apparently not a frequent issue, so it doesn't matter that much if it isn't fixed now.

Special care should be done by testing SW mode, as I had to adjust how timelimit is reset server side to fix a display issue in #495... (ugh). Fix seems okay, but if shit happens that could likely be there, and I'd rather not release another emergency hotfix again :P

**Reply by rmarquis** [reply]
The current situation becomes quite frustrating, so I'd suggest to make a last testing build, and release the hotfix this weekend or at the beginning of next week. The ongoing cup might give additional feedback on issues to fix, but if there isn't anything major we should get back to a normal dev activity asap.

What'd you think?

**Reply by BystryPL** [reply]
Since I'm in charge of the updating process on gather/scrim servers (passworded servers used for competitive games called "gathers" or "scrims") and you're not present on Discord, I thought on this occasion I'd give you a little insight of how the feedback/update process on ETL looks like these days so you don't stay completely in the dark as we don't discuss those things over here on GitHub.

Gather servers are nowadays the testing ground for new snapshots and 99% of feedback comes from the competitive scene (with some exceptions, like that bug causing clients to crash when objectives explode). TM public main server is usually lagging behind the gather servers a few snapshots; they update when they make sure that there are no game-breaking bugs. For instance - gather servers are currently on snapshot `44-g1380848` while TM main is still on `2.78` (updates are frozen on gather servers until the cup is over though). Normally we update gather servers when there are some changes that impact competitive gameplay and/or new features are added - I'd say on average once per week, depending on how active the development is in given period.

That being said, the on-going cup has nothing to do with the feedback process (and once again, the updates are frozen for the duration of the cup) - the majority of feedback comes from regular gathers (3on3 and 6on6 games that people queue for on ET: Legacy Scrims Discord server). The amount of "cup" games are small compared to the amount of regular gathers - as there are multiple gathers each day and that's why we have a tremendous amount of feedback thanks to them. To illustrate the numbers - in the `28.09.21 - 29.10.21` period, **316** 3on3 gathers took place:

![Zrzut ekranu 2021-10-29 o 10 58 11](https://user-images.githubusercontent.com/45846431/139408642-f99f6718-3d60-4a7a-b057-4dafdcc41e65.png)

And **70** 6on6 gathers:

![Zrzut ekranu 2021-10-29 o 11 08 30](https://user-images.githubusercontent.com/45846431/139408750-a24d8946-e70e-470a-8fd6-3c09ba7c0d18.png)

If I count correctly, that's an average of almost 13 games per day in the last 30 days.

Regarding the hotfix release, you may want to take into account that the gather servers won't be updated until after the cup is over (cup final is currently planned for 7th November) and TM main server _probably_ won't update before we do. So even if you release the hotfix ASAP, there probably will be no feedback anyway, at least on the mod side.

**Reply by rmarquis** [reply]
By any chance, do you have numbers from etpro that we can compare to Legacy gathers? Just being curious here.

**Reply by Helldembez** [reply]
3v3
![afbeelding](https://user-images.githubusercontent.com/11442728/139652274-c5d1f197-949c-47e1-8ab0-86b04ee6dc37.png)


6v6
![afbeelding](https://user-images.githubusercontent.com/11442728/139652250-e341f853-f053-4aaa-b714-63f53558d4b6.png)

**Comment by rmarquis** [comment]
Thank you for the detailed overview - I really appreciate it.

I mentioned the cup because that's were I noticed the upside down FT icon with 2.60b (some dinosaurs playing it...), but it makes sense that gathers give the most immediate feedback. I see the process seem reversed now (before TM public used to be the guinea pigs afaik).

As for the release, I'm just afraid we'll be stuck (again) in a "not enough tested, so delay it again" loop if we keep adding new stuff. 
Overall quality has always been an issue with release, but I'm at least glad the process improved greatly, especially the past year or so (thanks to TM and all your hard work).

It's rather annoying to wait 2 more weeks for that release hotfix, so I am not sure what to do. Wait again and don't go too crazy in the meantime, or we could release soon, test on our own servers with limited people and hope for the best. Suggestion welcome here.

**Reply by BystryPL** [reply]
What about freezing the commits until 7th November to ensure that nothing breaks and coordinate the hotfix release with the cup final?

**Reply by jackeri** [reply]
Or just do a release and if its fucked do another one? Making a release takes about an hour, so it should not be an issue.

**Reply by rmarquis** [reply]
Yes. We can revert 75c71ffd403956b9d11a41880cd0ca834284a666 and release now. Works for me.

**Reply by Aranud** [reply]
Revert, done. PR created.

**Reply by rmarquis** [reply]
It's the first of the month. Shall we finally get that release out of the way?
I just think we should merge the Android versioning fix (#1801) which *should* work correctly, unless git has a different behaviour on various Linux distro.

Who's pulling the trigger?

**Reply by Helldembez** [reply]
Kemon is probably going to do it , but could not find the time to do so on Sunday. I think because of some IRL stuff.

**Reply by rmarquis** [reply]
You probably meant jacker here :D
